closest_point_on_segment_xy returns a 3D point (Z corrdinate !=0)

**Describe the bug**

If the `closest_point_on_segment_xy` is one of the endpoints, it is returned with z coordinate different than 0.
This is because the [logic returns the endpoint as-is](https://github.com/compas-dev/compas/blob/961b15e7560d2b450e6caaa77ad5676af4103beb/src/compas/geometry/_core/distance.py#L766), without zeroing the Z coordinate.

**To Reproduce**

```py
from compas.geometry import Point, Line, closest_point_on_segment_xy


def test_closest_point_segment_xy():
    point = Point(0, 0, -10)
    line = Line([1, 1, -15], [10, 10, 20])
    ponl= closest_point_on_segment_xy(point, line)
    assert ponl == Point(1, 1, 0)
```

it returns

```
=============================== FAILURES ================================
_________________ test_closest_point_segment_xy ________________________

    def test_closest_point_segment_xy():
        point = Point(0, 0, -10)
        line = Line([1, 1, -15], [10, 10, 20])
        ponl = closest_point_on_segment_xy(point, line)
>       assert ponl == Point(1, 1, 0)
E       assert Point(1.000, 1.000, -15.000) == Point(1.000, 1.000, 0.000)
E         Use -v to get the full diff
```

--- a/AUTHORS.md
+++ b/AUTHORS.md
@@ -30,3 +30,4 @@
 - Xingxin He <<xingxin.he@mail.polimi.it>> [@XingxinHE](https://github.com/XingxinHE)
 - Robin Godwyll <<rodwyll+github@gmail.com>> [@robin-gdwl](https://github.com/robin-gdwl)
 - Mattis Koh <<mattiskoh@gmail.com>> [@mattiskoh](https://github.com/mattiskoh)
+- Andrea Ghensi <<a.ghensi@swsglobal.com>> [@sanzoghenzo](https://github.com/sanzoghenzo)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Changed directory where ghuser components are installed.
 * Added ghuser components directory to those removed by the `clean` task.
 * Clean up the ghuser directory before building ghuser components.
+* Fixed bug in `compas.geometry.distance.closest_point_on_segment_xy`
 
 ### Removed
 

--- a/src/compas/geometry/_core/distance.py
+++ b/src/compas/geometry/_core/distance.py
@@ -764,8 +764,8 @@ def closest_point_on_segment_xy(point, segment):
     d2 = distance_point_point_sqrd_xy(b, p)
     if d1 > d or d2 > d:
         if d1 < d2:
-            return a
-        return b
+            return [a[0], a[1], 0.0]
+        return [b[0], b[1], 0.0]
     return p
 
 

--- a/tests/compas/geometry/test_distance.py
+++ b/tests/compas/geometry/test_distance.py
@@ -1,10 +1,14 @@
+import pytest
+
 from compas.geometry import Point
 from compas.geometry import Line
 from compas.geometry import closest_point_on_segment_xy
 
 
-def test_closest_point_segment_xy():
-    point = Point(0, 0, -10)
+@pytest.mark.parametrize(
+    "point,point_on_line", [[[0, 0, -10], [1, 1, 0]], [[5, 4, 80], [4.5, 4.5, 0]]]
+)
+def test_closest_point_segment_xy(point, point_on_line):
     line = Line([1, 1, -15], [10, 10, 20])
-    ponl = closest_point_on_segment_xy(point, line)
-    assert ponl == Point(1, 1, 0)
+    ponl = closest_point_on_segment_xy(Point(*point), line)
+    assert ponl == Point(*point_on_line)

--- a/tests/compas/geometry/test_distance.py
+++ b/tests/compas/geometry/test_distance.py
@@ -1,0 +1,10 @@
+from compas.geometry import Point
+from compas.geometry import Line
+from compas.geometry import closest_point_on_segment_xy
+
+
+def test_closest_point_segment_xy():
+    point = Point(0, 0, -10)
+    line = Line([1, 1, -15], [10, 10, 20])
+    ponl = closest_point_on_segment_xy(point, line)
+    assert ponl == Point(1, 1, 0)

